libraries/AP_Notify/Buzzer.cpp

```diff
@@ -57,118 +57,23 @@ bool Buzzer::init()
 // update - updates led according to timed_updated.  Should be called at 50Hz
 void Buzzer::update()
 {
+    update_pattern_to_play();
+    update_playing_pattern();
+}
+
+void Buzzer::update_pattern_to_play()
+{
+    if (AP_HAL::millis() - _pattern_start_time < _pattern_start_interval_time_ms) {
+        // do not interrupt playing patterns / enforce minumum separation
+        return;
```

Comment:
If we want to keep old behavior, this should be below the arming_failed event. It does make sense to me that the event interrupts any other pattern being played - also realize that we only have one shot at playing it, events are cleared after each update call.

Reply:
At 32 bits each we could keep a short queue ;-)

Moved it.

Reply:
> At 32 bits each we could keep a short queue ;-)

We could, but by the time we would play a pattern it could not be valid anymore.

> Moved it.

Did you move it to where you really wanted to? If yes, why put it below the armed flag? That's a functional change from previous code.